### PR TITLE
added filename argument to run()

### PR DESCRIPTION
**Pull request by SanRehmo**
Added a `filename` argument to `run()` allowing to explicitly specify the output filename/path when running inference.

### CODE REVIEW

File: extra_model/_run.py [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
 
 MODELS_FOLDER = "./embeddings"
-OUTPUT_FILE = "result.csv"
+OUTPUT_FILE = Path("result.csv")
 
 logger = logging.getLogger(__name__)
 
 
-def run(input_path: Path, output_path: Path) -> None:
+def run(
```
Comment (romatik) [inline extra_model/_run.py:14]: You also need to add this parameter to `_cli.py`.
It is also a good idea to add an example of using this parameter to README.

**Review by romatik — CHANGES_REQUESTED** [review]

File: extra_model/_run.py (hunk 1 above)
Comment (romatik) [reply extra_model/_run.py:14]: Also, what happens if people don't put `.csv` in the end of the name? Do we want to enforce that?

Comment (SanRehmo) [reply extra_model/_run.py:14]: added the argument to `_cli.py`. On the second point, I would not enforce a file extension but curious to hear your opinions on this @romatik @chrisantonellis

Comment (romatik) [reply extra_model/_run.py:14]: it's a bit of a footgun, but given that we add info into the help and an example to the README, it should be alright.

File: extra_model/_cli.py [hunk 2]
```diff
@@ -12,24 +12,30 @@
...
+    input_path: Path, output_path: Path, output_filename: Path, debug: bool = False
+) -> None:
     """Run the Extra algorithm for unsupervised topic extraction.
 
     INPUT_PATH is the path to the input parquet file with the user generated texts.
-    
+
     OUTPUT_PATH is the path to the output directory. Default is `/io`.
+
+    OUTPUT_FILENAME is the filename of the output file. Default is `result.csv`
```
Comment (romatik) [inline extra_model/_cli.py:33]: Mention here that we don't enforce `.csv`

File: extra_model/_run.py [hunk 3]
```diff
@@ -26,4 +28,4 @@ def run(input_path: Path, output_path: Path) -> None:
     if not output_path.exists():
         output_path.mkdir(parents=True)
 
-    results.to_csv(output_path / OUTPUT_FILE, encoding="utf-8", index=False)
+    results.to_csv(output_path / output_filename, encoding="utf-8", index=False)
```
Comment (romatik) [inline extra_model/_run.py:31]: I'm not sure why it's not covered by the tests, but if that's the only problem with the PR, I'm OK with approving.

File: extra_model/_cli.py (hunk 2 above)
Comment (SanRehmo) [reply extra_model/_cli.py:33]: done

File: extra_model/_run.py (hunk 3 above)
Comment (SanRehmo) [reply extra_model/_run.py:31]: Yeah, not sure either. I've adjusted the relevant entrypoint test accordingly [here](https://github.com/wayfair-incubator/extra-model/blob/cr105s_add_output_filename_argument/tests/test_cli.py#L64).

**Review by romatik — APPROVED** [review]

**Pull request merged**